neurodamus/core/coreneuron_configuration.py

```diff
@@ -0,0 +1,129 @@
+                float(end_time),
+                num_gids,
+                buffer_size
+            ])) + "\n").encode())
+            # Write the array of integers to the file in binary format
+            fp.write(struct.pack(f'{num_gids}i', *gids))
+            fp.write(b'\n')
+
+    @run_only_rank0
+    def write_sim_config(
+            self, tstop, dt, forwardskip, prcellgid, celsius, v_init,
+            pattern=None, seed=None, model_stats=False, enable_reports=True):
+        simconf = Path(self.output_root) / self.sim_config_file
+        logging.info(f"Writing sim config file: {simconf}")
+        simconf.parent.mkdir(parents=True, exist_ok=True)
+        with simconf.open("w") as fp:
+            fp.write(f"outpath='{os.path.abspath(self.output_root)}'\n")
+            fp.write(f"datpath='{os.path.abspath(self.datadir)}'\n")
+            fp.write(f"tstop={tstop}\n")
+            fp.write(f"dt={dt}\n")
+            fp.write(f"forwardskip={forwardskip}\n")
+            fp.write(f"prcellgid={int(prcellgid)}\n")
+            fp.write(f"celsius={celsius}\n")
+            fp.write(f"voltage={v_init}\n")
+            fp.write(f"cell-permute={int(self.default_cell_permute)}\n")
+            if pattern:
+                fp.write(f"pattern='{pattern}'\n")
+            if seed:
+                fp.write(f"seed={int(seed)}\n")
+            if model_stats:
+                fp.write("'model-stats'\n")
+            if enable_reports:
+                fp.write(f"report-conf='{self.output_root}/{self.report_config_file}'\n")
+            fp.write("mpi=true\n")
+
+    @run_only_rank0
+    def write_report_count(self, count):
+        report_config = Path(self.output_root) / self.report_config_file
+        report_config.parent.mkdir(parents=True, exist_ok=True)
+        with report_config.open("a") as fp:
+            fp.write(f"{count}\n")
+
+    @run_only_rank0
+    def write_population_count(self, count):
+        self.write_report_count(count)
+
+    @run_only_rank0
+    def write_spike_population(self, population_name, population_offset=None):
+        report_config = Path(self.output_root) / self.report_config_file
+        report_config.parent.mkdir(parents=True, exist_ok=True)
+        with report_config.open("a") as fp:
+            fp.write(population_name)
+            if population_offset is not None:
+                fp.write(f" {int(population_offset)}")
+            fp.write("\n")
+
+    @run_only_rank0
+    def write_spike_filename(self, filename):
+        report_config = Path(self.output_root) / self.report_config_file
+        report_config.parent.mkdir(parents=True, exist_ok=True)
+        with report_config.open("a") as fp:
+            fp.write(filename)
+            fp.write("\n")
+
+    def psolve_core(self, save_path=None, restore_path=None):
+        from neuron import h
+        h.CVode().cache_efficient(1)
+        from neuron import coreneuron
+        from . import NeurodamusCore as Nd
```

Comment:
In principle `Nd` is already a wrapper over `Neuron.h`, we can in principle avoid importing the latter